sprockets/mixins/amqp/__init__.py

```diff
@@ -254,24 +238,21 @@ def _on_channel_close(self, channel, reply_code, reply_text):
         LOGGER.warning('RabbitMQ closed the channel (%s): %s',
                        reply_code, reply_text)
         if not self._connecting:
-            LOGGER.info('Reopening RabbitMQ channel...')
+            self._channel = None
+            self._connecting = True
             self._open_channel()
 
-    @property
-    def _parameters(self):
-        """Return a pika URLParameters object using the :envvar:`AMQP_URL`
-        environment variable which identifies the Rabbit MQ server as a URL
-        ready for :class:`pika.connection.URLParameters`.
 
-        :rtype: pika.URLParameters
+def install(application, io_loop=None, **kwargs):
+    """Call this to install AMQP for the Tornado application."""
+    if getattr(application, 'amqp', None) is not None:
+        LOGGER.warning('AMQP is already installed')
+        return False
 
-        """
-        return pika.URLParameters(os.environ.get('AMQP_URL', self.DEFAULT_URL))
+    if os.environ.get('AMQP_TIMEOUT'):
+        kwargs['timeout'] = int(os.environ['AMQP_TIMEOUT'])
+    if os.environ.get('AMQP_URL'):
+        kwargs['url'] = os.environ['AMQP_URL']
 
-    @gen.coroutine
-    def _wait_for_connection(self):
-        """Wait for a pending AMQP connection to complete"""
-        try:
-            yield self._ready.wait(self._timeout)
-        except gen.TimeoutError:
-            self._connection_timeout()
+    setattr(application, 'amqp', AMQP(**kwargs))
```

Comment:
You should be passing `io_loop` through here.  I'd remove it from the parameter list and let `kwargs` catch it.